libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationEventDispatcher.java

```diff
@@ -131,6 +134,24 @@ void removeFasterRouteListener(@Nullable FasterRouteListener fasterRouteListener
     }
   }
 
+  void addRawLocationListener(@NonNull RawLocationListener rawLocationListener) {
+    if (rawLocationListeners.contains(rawLocationListener)) {
+      Timber.w("The specified RawLocationListener has already been added to the stack.");
+      return;
+    }
+    rawLocationListeners.add(rawLocationListener);
+  }
+
+  void removeRawLocationListener(@Nullable RawLocationListener rawLocationListener) {
+    if (rawLocationListener == null) {
+      rawLocationListeners.clear();
```

Comment:
Is there a reason for clearing when null is passed? If purposely inconsistent with the way the other listeners are handled, should this maybe be named different?

Reply:
> the other listeners are handled

@devotaaabel do you mean with other listeners in general?  I used this same logic copied from the other listeners in `MapboxNavigation`